docs/develprocess.md

```diff
@@ -858,6 +858,91 @@ genisoimage -o $OUTPUT -v -r -J --joliet-long -V "XCP-ng $VERSION" -c boot/isoli
 isohybrid --uefi $OUTPUT
 ```
 
+## Create unattended ISO file
+
+This guide will provide an example steps on how to create unattended ISO image that can be used for mass deployment.
+
+### Prerequisites
+1. Create an answerfile
+```
+<?xml version="1.0"?>
+    <installation mode="fresh" csrtype="ext">
+        <primary-disk>sda</primary-disk>
+        <keymap>us</keymap>
+        <root-password type="plain">123456</root-password>
+        <hostname>server1.lab</hostname>
+        <bootloader location="partition">grub2</bootloader>
+        <network-backend>vswitch</network-backend>
+        <source type="local"/>
+        <admin-interface name="eth0" proto="static">
+            <ipaddr>192.168.1.10</ipaddr>
+            <subnet>255.255.255.0</subnet>
+            <gateway>192.168.1.1</gateway>
+        </admin-interface>
+        <name-server>192.168.1.1</name-server>
+        <name-server>8.8.8.8</name-server>
+        <ntp-server>192.168.1.1</ntp-server>
+        <ntp-server>192.168.1.2</ntp-server>
+        <timezone>Europe/Sofia</timezone>
+    </installation>
+```
+2. Download XCP-NG ISO
+
+### Image baking
+
+1. [Extract ISO](https://xcp-ng.org/docs/develprocess.html#extract-an-existing-iso-image)
+2. [Extact install.img installer into install directory](https://xcp-ng.org/docs/develprocess.html#extract-install-img)
+3. Copy answerfile to the installer
+```
+cp answerfile "$WORK_DIR/install/answerfile.xml"
+```
+4. Pack the installer image and cleanup install directory
+```
+cd $WORKDIR/install
+find . |  cpio -o -H newc | bzip2 > ../install.img
+cd ..
+rm $WORK_DIR/install -rf
+```
+5. Modify grub to use answerfile
+- BIOS - open $WORK_DIR/boot/isolinux/isolinux.cfg
+Change this
+```
+LABEL install
+        KERNEL mboot.c32
+        APPEND /boot/xen.gz dom0_max_vcpus=1-16 dom0_mem=max:8192M com1=115200,8n1 console=com1,vga --- /boot/vmlinuz console=hvc0 console=tty0 --- /install.img
+```
+Into this
+```
+LABEL install
+        KERNEL mboot.c32
+        APPEND /boot/xen.gz dom0_max_vcpus=1-16 dom0_mem=max:8192M com1=115200,8n1 console=com1,vga --- /boot/vmlinuz console=hvc0 console=tty0 answerfile=file:///answerfile.xml install --- /install.img
+```
+- UEFI - open $WORK_DIR/EFI/xenserver/grub.cfg
+change this:
+```
+menuentry "install" {
+    multiboot2 /boot/xen.gz dom0_max_vcpus=1-16 dom0_mem=max:8192M com1=115200,8n1 console=com1,vga
+    module2 /boot/vmlinuz xencons=hvc console=hvc0 console=tty0
+    module2 /install.img
+}
+```
+into this:
+```
+menuentry "install" {
+    multiboot2 /boot/xen.gz dom0_max_vcpus=1-16 dom0_mem=max:8192M com1=115200,8n1 console=com1,vga
+    module2 /boot/vmlinuz xencons=hvc console=hvc0 console=tty0 answerfile=file:///answerfile.xml install
+    module2 /install.img
+}
+```
```

Comment:
We should give users two different ways to do it:
1. Downloading the answerfile from the network: `answerfile=http://local_server/path/to/answerfile.xml`. This should be the first solution offered.
  * no need to modify install.img
  * you can change the configuration 
  * you can even have the answerfile served by a PHP or other script that will provide configuration that depends on the server (based on IP or mac address for example).
2. Embedding the answerfile into the ISO.

Reply:
Hey @stormi, can you give an example of how it can be done with PHP?
Currently i am adding the http example but i dont know how the PHP is done in practice.

Reply:
I was mentioning it as an advantage but I didn't intend to write documentation about it. It all differs on the situation.

Reply:
Alright. :)